src/worker/ingestion/process-event.ts

```diff
@@ -37,6 +37,32 @@ const MAX_FAILED_PERSON_MERGE_ATTEMPTS = 3
 // for e.g. internal events we don't want to be available for users in the UI
 const EVENTS_WITHOUT_EVENT_DEFINITION = ['$$plugin_metrics']
 
+// used to prevent identify from being used with generic IDs
+// that we can safely assume stem from a bug or mistake
+const DISALLOWED_IDS = new Set([
+    'anonymous',
+    'guest',
+    'not_authenticated',
+    'distinctId',
+    'distinct_id',
+    'id',
+    'ID',
+    '',
+    ' ',
+    0,
```

Comment:
If the type of `distinctId` is right in `EventsProcessor` – `string`, then this will never match, as `0` is a `number`.

Reply:
ah yes sorry accidental typo